src/compiler/crystal/tools/doc/generator.cr

```diff
@@ -414,4 +414,13 @@ class Crystal::Doc::Generator
     end
     locations.sort
   end
+
+  def highlight(code)
```

Comment:
This method is used in many places. Can we instead make `Highlighter.highlight` surround everything with a begin/rescue and just return `code` on exception?

Reply:
No. I think `Highlighter#.highlight` should not have responsibility to show warning message. See the below TODO comment.

/cc @straight-shoota

Reply:
Yeah I'd say this makes more sense here because the generator can decide how to deal with an error. Defaulting to non-highlighted code is fine. But as explained in https://github.com/crystal-lang/crystal/pull/9414#issuecomment-638278293, we could later add warnings or a strict mode that fails on error, as a means to catch mistakes.

Reply:
Yeah, and duplicating the code to generate the warning is not ideal. `Highlight.highlight` **is** the interface for this. If it needs a node to be passed to show the warning, then let's add it. But since the warning is not done in this PR, I think this is YAGNI.

Reply:
> Yeah, and duplicating the code to generate the warning is not ideal.

It's right. Thank you. However, when `Highlight.highlight` ignores an error on its processing, [highlighter's spec become poor](https://github.com/crystal-lang/crystal/pull/9416/commits/880b4c2b72a90d1ecaf9090549941d1e3a9e219d#diff-3a4c797455b77e25fbb8d7cd4df462dcL113-R114) in fact. But, I'm not sure these detailed tests are needed.